one-ui/ui/src/components/entities/mapping/mapping-card.tsx

```diff
@@ -57,6 +58,13 @@ const MappingCard: React.FC<Props> = (props) => {
     //For storing  mapping functions
     const [mapFunctions,setMapFunctions] = useState({});
 
+    //For storing namespaces
+    const [namespaces, setNamespaces] = useState({});
+    let nmspaces: any = {};
+    let mapIndexLocal: number = -1;
+    const [mapIndex, setMapIndex] = useState(-1);
+    let namespaceString = '';
```

Comment:
Trying to understand what these variables mean and how they are different from each other-- namespaces, nmspaces and namespaceString. Can you pls explain.

With all of us trying to write unit test, I feel its important we use standard naming conventions so its easy to read and understand the code. `nmspc`, `nmspaces`, `namespaces`, `ind`, `sDta`, `mData`, `mapArt` and `propty` are some examples.

Reply:
nmspaces - Local variable to store namespaces when iterating over  xml source doc. This will be used locally at other places as well.
namespaces - This is a state variable and will be passed on to the child component and have state management properties.
namespaceString - Just a string local variable to store the most recent namespace at a given time for any xml node. This is only a local temporary variable and has no significance for child component.

Any variable which is not part of the react state is a local variable for that component and has no relation with the child component.